telluric/vrt.py

```diff
@@ -121,8 +121,12 @@ def boundless_vrt_doc(
 
     vrt = BaseVRT(width, height, src_dataset.crs, transform)
 
-    for bidx, ci, block_shape, dtype in zip(src_dataset.indexes, src_dataset.colorinterp,
-                                            src_dataset.block_shapes, src_dataset.dtypes):
+    bidxs = src_dataset.indexes if bands is None else bands
+    for bidx in bidxs:
```

Comment:
how this code behaves when we have mask?

If it is not too difficult add a test the shows that mask is preserved